lib/fog/openstack/models/image_v2/images.rb

```diff
@@ -18,9 +18,9 @@ def summary(options = {})
           end
 
           def find_by_id(id)
-            image_hash = service.get_image_by_id(id).body
-            Fog::Image::OpenStack::V2::Image.new(
-                image_hash.merge(:service => service))
+            result = all.find { |image| image.id == id }
```

Comment:
Why not use the pattern that is common throughout fog-openstack? Something like this (for the Identity V2 tenant object, fog-openstack/lib/fog/identity/openstack/v2/models/tenants.rb):

```
          def find_by_id(id)
            cached_tenant = find { |tenant| tenant.id == id }
            return cached_tenant if cached_tenant
            tenant_hash = service.get_tenant(id).body['tenant']
            Fog::Identity::OpenStack::V2::Tenant.new(
              tenant_hash.merge(:service => service)
            )
          end
```